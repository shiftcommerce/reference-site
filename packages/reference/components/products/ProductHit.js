// Libraries
import { Component } from 'react'
import classNames from 'classnames'
import Link from 'next/link'

// Objects
import Image from '../../objects/Image'

class ProductHit extends Component {
  renderRatingStars (hit) {
    return (
      [1, 2, 3, 4, 5].map((key, index) =>
        <span key={key} className={classNames({'c-product__rating--fill': key <= hit.rating, 'c-product__rating--empty': key > hit.rating})}>
          &#9733;
        </span>
      )
    )
  }

  render () {
    let {
      hit
     } = this.props

    return (
      <section className='c-product'>
        <Link href='/products/708a0142-e60f-4332-9a8b-1359c5af9ec4' >
          <a>
            <div className='c-product__image-container'>
              <Image className='c-product__image' src={hit.image} />
            </div>

            <div className='c-product__title'>
              {hit.name}
            </div>

            <div className='c-product__price'>
          &pound;{hit.price}
            </div>

            <div>
              { this.renderRatingStars(hit) }
              <span className='c-product__rating-badge'>
                {hit.rating}
              </span>
            </div>
          </a>
        </Link>
      </section>
    )
  }
}

export default ProductHit
