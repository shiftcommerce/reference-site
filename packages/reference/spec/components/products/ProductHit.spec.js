// Component
import ProductHit from '../../../components/products/ProductHit'

// Fixtures
import productSearchHit from '../../fixtures/productSearchHit.fixture'

test('renders correctly', () => {
  // Arrange & Act
  const wrapper = mount(
    <ProductHit hit={productSearchHit} />
  )

  // Assert
  expect(wrapper).toMatchSnapshot()
  expect(wrapper).toIncludeText(productSearchHit.name)
  expect(wrapper).toIncludeText(productSearchHit.price)
})
