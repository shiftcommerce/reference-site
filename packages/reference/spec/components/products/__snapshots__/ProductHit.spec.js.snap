// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ProductHit
  hit={
    Object {
      "_highlightResult": Object {
        "category": Object {
          "matchLevel": "none",
          "matchedWords": Array [],
          "value": "Bathroom",
        },
        "description": Object {
          "matchLevel": "none",
          "matchedWords": Array [],
          "value": "Easy to move around with the included casters. Removable shelves which are easy to clean.",
        },
        "name": Object {
          "matchLevel": "none",
          "matchedWords": Array [],
          "value": "GRUNDTAL",
        },
        "sub_category": Object {
          "matchLevel": "none",
          "matchedWords": Array [],
          "value": "Bathroom > Bathroom storage",
        },
        "type": Object {
          "matchLevel": "none",
          "matchedWords": Array [],
          "value": "Cart, stainless steel",
        },
      },
      "_snippetResult": Object {
        "description": Object {
          "matchLevel": "none",
          "value": "Easy to move around with the included casters. Removable shelves",
        },
      },
      "category": "Bathroom",
      "colors": Array [],
      "description": "Easy to move around with the included casters. Removable shelves which are easy to clean.",
      "ikea_id": "60171433",
      "image": "http://www.ikea.com/us/en/images/products/grundtal-cart__26887_PE103191_S4.JPG",
      "materials": Array [
        "Stainless steel",
        "ABS plastic",
      ],
      "name": "GRUNDTAL",
      "objectID": "509663980",
      "price": 59.99,
      "rating": 2,
      "share_count": 41,
      "sizes": Array [
        "18 7/8x9 1/2x30 3/8\\"",
      ],
      "sub_category": "Bathroom > Bathroom storage",
      "type": "Cart, stainless steel",
    }
  }
>
  <section
    className="c-product"
  >
    <Link
      as="/products/708a0142-e60f-4332-9a8b-1359c5af9ec4"
      href="/products/product?id=708a0142-e60f-4332-9a8b-1359c5af9ec4"
    >
      <a
        href="/products/708a0142-e60f-4332-9a8b-1359c5af9ec4"
        onClick={[Function]}
      >
        <div
          className="c-product__image-container"
        >
          <Image
            className="c-product__image"
            src="http://www.ikea.com/us/en/images/products/grundtal-cart__26887_PE103191_S4.JPG"
          >
            <div
              className="c-product__image"
              style={
                Object {
                  "height": undefined,
                  "width": undefined,
                }
              }
            >
              <img
                className="c-image"
                src="http://www.ikea.com/us/en/images/products/grundtal-cart__26887_PE103191_S4.JPG"
              />
            </div>
          </Image>
        </div>
        <div
          className="c-product__title"
        >
          GRUNDTAL
        </div>
        <div
          className="c-product__price"
        >
          £
          59.99
        </div>
        <div>
          <span
            className="c-product__rating--fill"
          >
            ★
          </span>
          <span
            className="c-product__rating--fill"
          >
            ★
          </span>
          <span
            className="c-product__rating--empty"
          >
            ★
          </span>
          <span
            className="c-product__rating--empty"
          >
            ★
          </span>
          <span
            className="c-product__rating--empty"
          >
            ★
          </span>
          <span
            className="c-product__rating-badge"
          >
            2
          </span>
        </div>
      </a>
    </Link>
  </section>
</ProductHit>
`;
